coffeescripts/xwing.coffee

```diff
@@ -758,9 +770,22 @@ class exportObj.SquadBuilder
         $(window).resize =>
             @select_simple_view_button.click() if $(window).width() < 768 and @list_display_mode != 'simple'
 
-        @notes.change @onNotesUpdated
+        # separate function for on change that accesses the notes values
+        @notes.change @onNotesChanged
         @notes.on 'keyup', @onNotesUpdated
 
+    updatePermaLink: () =>
+        squad_link = "#{window.location.href.split('?')[0]}?f=#{encodeURI @faction}&d=#{encodeURI @serialize()}&sn=#{encodeURIComponent @current_squad.name}"
+        if @current_squad.additional_data.notes
+            squad_link = squad_link + "&sno=#{encodeURIComponent @current_squad.additional_data.notes}"
+        @permalink.attr 'href', squad_link
+
+    onNotesChanged: =>
+        if @total_points > 0
```

Comment:
There's a corner case here where if the user starts with a new squad, updates the notes first, and then adds ships, the notes don't get saved (because `total_points` was 0 at the time).